r/tests/testthat/test-Array.R

```diff
@@ -819,3 +819,23 @@ test_that("auto int64 conversion to int can be disabled (ARROW-10093)", {
     expect_true(inherits(as.data.frame(batch)$x, "integer64"))
   })
 })
+
+
+test_that("Array to C-interface", {
```

Comment:
These tests are good, now we have test coverage for the C bridge without requiring pyarrow.

Very tangentially related: what if we changed py_to_r and r_to_py to use export_to_c and import_from_c? Functionally it's the same but I wonder if it reads better since there's more harmony with the pyarrow methods.

Reply:
Yeah, I thought about spinning up a new subprocess of R to pass them back and forth, but this works and is pretty quick and much less complicated to test the functions — and the pyarrow stuff tests the nested process passing already.

I thought about that, and can totally do it — when I did it before with one of the functions I abandoned that path because it didn't make the code any shorter since these wrappers are pretty thing, but let me try again and see how it reads. The harmony/parallelism is probably worth it even if it's not shorter.